AzureCore/Source/Pipeline/Policies/LoggingPolicy.swift

```diff
@@ -32,7 +32,7 @@ public class LoggingPolicy: PipelineStageProtocol {
                                                            options: .caseInsensitive)
     public init() {}
 
-    public func onRequest(_ request: inout PipelineRequest) {
+    public func onRequest(_ request: PipelineRequest, then completion: @escaping (PipelineRequest) -> Void) {
```

Comment:
Why is this one marked `@escaping` when the others aren't?

Also might be good for ergonomics since they appear often, to have type aliases for these completion handlers. Something like `PipelineCompletionHandler` and `PipelineErrorHandler`?

Reply:
They should all be escaping. Good point on the typealias. I will go that route.